engine/runtime-language-epb/src/main/java/org/enso/interpreter/epb/PyForeignNode.java

```diff
@@ -39,10 +33,68 @@ public abstract class PyForeignNode extends ForeignFunctionCallNode {
   private InteropLibrary nodePythonZone;
   @CompilerDirectives.CompilationFinal
   private Object fnPythonCombine;
+  @CompilerDirectives.CompilationFinal
+  private Object none;
   @Child
   private InteropLibrary nodePythonCombine;
+  @Child
+  private InteropLibrary iop = InteropLibrary.getFactory().createDispatched(3);
 
-  abstract Object getForeignFunction();
+  PyForeignNode(CallTarget ct) {
+    super(ct);
+  }
+
+  @Override
+  public Object execute(Object[] arguments) throws InteropException {
+    for (int i = 0; i < arguments.length; i++) {
+      var javaTime = iop.isTime(arguments[i]) ? iop.asTime(arguments[i]) : null;
+      var time = javaTime != null ? wrapPythonTime(javaTime) : null;
+      var javaDate = iop.isDate(arguments[i]) ? iop.asDate(arguments[i]) : null;
+      var date = javaDate != null ? wrapPythonDate(javaDate) : null;
+      var zone = iop.isTimeZone(arguments[i]) ? wrapPythonZone(iop.asTimeZone(arguments[i]), javaTime, javaDate) : null;
+      if (date != null && time != null) {
+        arguments[i] = combinePythonDateTimeZone(date, time, zone);
+      } else if (date != null) {
+        arguments[i] = date;
+      } else if (time != null) {
+        arguments[i] = time;
+      } else if (zone != null) {
+        arguments[i] = zone;
+      }
+      // Enso's Text type should be converted to TruffleString before sent to python, because
+      // python does not tend to use InteropLibrary, so any Text object would be treated as
+      // a foreign object, rather than 'str' type.
+      if (iop.isString(arguments[i])) {
+        arguments[i] = iop.asTruffleString(arguments[i]);
+      }
+    }
+    var res = super.execute(arguments);
+    if (iop.hasMetaObject(res)) {
+      var meta = iop.getMetaObject(res);
+      var name = iop.getMetaQualifiedName(meta);
+      if ("module".equals(name)) {
```

Comment:
Workaround to support [make_null](https://github.com/enso-org/enso/blob/9fb3d9a2fa9e618c4052c15dcaff49bb6348b011/test/Tests/src/Semantic/Python_Interop_Spec.enso#L74) - e.g. `return None` which currently returns its module rather than `None`.

Reply:
So we can't use `iop.isNull` ?